File: backend/app/models/meal.rb
class Meal < ApplicationRecord
  belongs_to :user
  has_many :meal_ingredients, dependent: :destroy
  has_many :ingredients, through: :meal_ingredients
  has_many :meal_dishes, dependent: :destroy
  has_many :dishes, through: :meal_dishes

  validates :meal_date, presence: true
  validates :category, presence: true
  validates :user_id, presence: true

  enum category: { 
    breakfast: 'breakfast', 
    lunch: 'lunch', 
    dinner: 'dinner', 
    snack: 'snack' 
  }
end

File: graphqlapi/src/backend/domain/entities/Meal.ts
// src/backend/domain/entities/Meal.ts
import { DishWithIngredients } from "@/backend/domain/entities/Dish";
import { MealDish } from "@/backend/domain/entities/MealDish";
export interface Meal {
  id: number;
  userId: number;
  mealDate: Date;
  startTime: Date; // prismaの仕様
  endTime: Date; // prismaの仕様
  category: string; // 'breakfast', 'lunch', 'dinner', 'snack'
  createdAt: Date;
  updatedAt: Date;
  // 関連エンティティ（オプション、循環参照を避ける）
  dishIds?: number[];
}

export interface MealWithDishes extends Meal {
  mealDishes: MealDishWithDish[];
  dishes: DishWithIngredients[];
}

export interface MealDishWithDish extends Meal {
  dish: DishWithIngredients;
}

export interface UpdateMealRepositoryInput {
  mealDate: Date;
  category: string;
  startTime: string;
  endTime: string;
}

export interface CreateMealRepositoryInput {
  userId: number;
  mealDate: Date;
  category: string;
  startTime: string;
  endTime: string;
}
